include/aws/common/linked_list.h

```diff
@@ -212,6 +212,34 @@ AWS_STATIC_IMPL void aws_linked_list_insert_after(
     AWS_POSTCONDITION(after->next == to_add);
 }
 
+/**
+ * Swaps the order two nodes in the linked list.
+ */
+AWS_STATIC_IMPL void aws_linked_list_swap_nodes(struct aws_linked_list_node *a, struct aws_linked_list_node *b) {
+    AWS_PRECONDITION(aws_linked_list_node_next_is_valid(a));
+    AWS_PRECONDITION(aws_linked_list_node_next_is_valid(b));
+    AWS_PRECONDITION(a != NULL);
+    AWS_PRECONDITION(b != NULL);
+
+    struct aws_linked_list_node *a_prev_cpy = a->prev;
+    struct aws_linked_list_node *a_prev_next = a->next;
+
```

Comment:
Function fails to ever update the value of `a->prev`

Can I be an overstepping douchebag and suggest this implementation instead?

```
// swap A and B
tmp = *a
*a = *b
*b = tmp

// fix up adjacent nodes
a->prev->next = a
a->next->prev = a

b->prev->next = b
b->next->prev = b
```

Reply:
I'm a dummy. This doesn't work if A and B are adjacent.

Reply:
ok, here's my new proposal. different branches for adjacent swap vs non-adjacent swap
```
if (a == b)
    return

// test if A and B are adjacent
first_adjacent = NULL
if (a->next == b)
    first_adjacent = a
else if (b->next == a)
    first_adjacent = b

if (first_adjacent)
    // number nodes by the order we WANT them to be
    node1 = first_adjacent->prev
    node2 = first_adjacent->next
    node3 = first_adjacent
    node4 = first_adjacent->next->next

    // now put them in that order
    node1->next = node2

    node2->prev = node1
    node2->next = node3

    node3->prev = node2
    node3->next = node4

    node4->prev = node3

else
    // swap non-adjacent A and B
    tmp = *a
    *a = *b
    *b = tmp

    // fix up nodes adjacent to A and B
    a->prev->next = a
    a->next->prev = a

    b->prev->next = b
    b->next->prev = b
```